model/navigation/LtiNavigationService.php

```diff
@@ -127,23 +136,28 @@ protected function shouldShowThankYou(LtiLaunchData $launchData): bool
         return $this->getOption(self::OPTION_THANK_YOU_SCREEN);
     }
 
-    /**
-     * @return string
-     */
     protected function buildThankYouUrl(): string
     {
-        return $this->getServiceLocator()->get(UrlHelper::class)->buildUrl(
-            'thankYou',
+        return $this->getUrlHelper()->buildUrl('thankYou', 'DeliveryRunner', 'ltiDeliveryProvider');
+    }
+
+    protected function buildFeedbackUrl(DeliveryExecutionInterface $deliveryExecution): string
+    {
+        return $this->getUrlHelper()->buildUrl(
+            'feedback',
             'DeliveryRunner',
-            'ltiDeliveryProvider'
+            'ltiDeliveryProvider',
+            [
+                'deliveryExecution' => $deliveryExecution->getIdentifier(),
+            ]
```

Comment:
I would grab the reason from here https://github.com/oat-sa/extension-tao-ltideliveryprovider/pull/365/files#diff-383263564fa2f0d212455a616a70ee16cb9d7cf20c0a5bd9d9b42cc737a5dd33R65 and forward by parameter, rather than evaluate it in multiple places

```suggestion
            'ltiDeliveryProvider',
            [
                'deliveryExecution' => $deliveryExecution->getIdentifier(),
                'reason' => 'someReasonIdHere'
            ]
```